### PR TITLE
Basic login and register functionality

### PR DESCRIPTION
**Pull request by Anntol**
Done:
- Basic register, login and logout functionality to be able to use loggedUserId in another API requests

TODO:
- handling errors
- refresh token
- adding photo

### CODE REVIEW

File: rslang/src/app/auth/components/login/login.component.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<section class="auth-page">
+  <div class="auth-page__wrapper">
+    <mat-card>
+      <form #loginForm="ngForm" (submit)="onSubmit(loginForm)">
+        <mat-form-field>
+          <input matInput name="email" type="email" placeholder="Email" ngModel #emailInput="ngModel" required email />
```
Comment (Nikitzu) [inline rslang/src/app/auth/components/login/login.component.html]: Cant we use more common ngModel syntax?

File: rslang/src/app/auth/components/login/login.component.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          <mat-error *ngIf="emailInput.invalid">Please enter a valid email.</mat-error>
+        </mat-form-field>
+        <mat-form-field>
+          <input
+            matInput
+            name="password"
+            type="password"
+            placeholder="Password"
+            ngModel
```
Comment (Nikitzu) [inline rslang/src/app/auth/components/login/login.component.html]: same

File: rslang/src/app/auth/components/register/register.component.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<section class="auth-page">
+  <div class="auth-page__wrapper">
+    <mat-card>
+      <form #authForm="ngForm" (submit)="onSubmit(authForm)">
+        <mat-form-field>
+          <input matInput name="name" type="text" placeholder="Name" ngModel #nameInput="ngModel" required />
```
Comment (Nikitzu) [inline rslang/src/app/auth/components/register/register.component.html]: same

File: rslang/src/app/auth/components/register/register.component.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+          <mat-error *ngIf="nameInput.invalid">Please enter a valid name.</mat-error>
+        </mat-form-field>
+        <mat-form-field>
+          <input matInput name="email" type="email" placeholder="Email" ngModel #emailInput="ngModel" required email />
```
Comment (Nikitzu) [inline rslang/src/app/auth/components/register/register.component.html]: same

File: rslang/src/app/auth/components/register/register.component.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+          <mat-error *ngIf="emailInput.invalid">Please enter a valid email.</mat-error>
+        </mat-form-field>
+        <mat-form-field>
+          <input
+            matInput
+            name="password"
+            type="password"
+            placeholder="Password"
+            ngModel
```
Comment (Nikitzu) [inline rslang/src/app/auth/components/register/register.component.html]: same

File: rslang/src/app/auth/interceptors/auth.interceptor.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { AuthService } from '../services/auth.service';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  constructor(private authService: AuthService) {}
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (Nikitzu) [inline rslang/src/app/auth/interceptors/auth.interceptor.ts]: ?

File: rslang/src/app/auth/interceptors/auth.interceptor.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
```
Comment (Nikitzu) [inline rslang/src/app/auth/interceptors/auth.interceptor.ts]: why any?

File: rslang/src/app/auth/services/auth-action.service.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+@Injectable()
+export class AuthActionService extends BaseActionService {
+  constructor(
+    httpClient: HttpClient,
+    private authService: AuthService,
+  ) {
+    super(httpClient);
+  }
+
+  createUser(newUser: CreateUser, authService = this.authService): void {
```
Comment (Nikitzu) [inline rslang/src/app/auth/services/auth-action.service.ts]: Bad type name. CreateUser is a method not a model

File: rslang/src/app/auth/services/auth-action.service.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    const action: CallbackObject = {
+      onSuccess() {
+        authService.redirectToUrl('/auth');
+      },
+      // TODO catch error
+    };
+    this.sendAction('POST', `${BASE_URL}/users`, action, { body: newUser });
+  }
+
+  signinUser(user: SigninRequest, authService = this.authService): void {
```
Comment (Nikitzu) [inline rslang/src/app/auth/services/auth-action.service.ts]: why SigninRequest property named user?

File: rslang/src/app/components/dot-menu/dot-menu.component.ts [hunk 10]
```diff
@@ -1,7 +1,32 @@
...
 
 @Component({
   selector: 'app-dot-menu',
   templateUrl: './dot-menu.component.html',
 })
-export class DotMenu {}
+export class DotMenu implements OnInit, OnDestroy {
+  isUserAuthenticated = false;
+
+  private authStatusSubscriber!: Subscription;
```
Comment (Nikitzu) [inline rslang/src/app/components/dot-menu/dot-menu.component.ts]: authStatusSubscribtion?

**Review by Nikitzu — APPROVED** [review]

File: rslang/src/app/auth/components/login/login.component.html (hunk 1 above)
Comment (Anntol) [reply rslang/src/app/auth/components/login/login.component.html]: changed to ReactiveForms

File: rslang/src/app/auth/interceptors/auth.interceptor.ts (hunk 7 above)
Comment (Anntol) [reply rslang/src/app/auth/interceptors/auth.interceptor.ts]: changed to unknown

File: rslang/src/app/auth/services/auth-action.service.ts (hunk 8 above)
Comment (Anntol) [reply rslang/src/app/auth/services/auth-action.service.ts]: fixed

File: rslang/src/app/auth/services/auth-action.service.ts (hunk 9 above)
Comment (Anntol) [reply rslang/src/app/auth/services/auth-action.service.ts]: fixed

File: rslang/src/app/components/dot-menu/dot-menu.component.ts (hunk 10 above)
Comment (Anntol) [reply rslang/src/app/components/dot-menu/dot-menu.component.ts]: fixed

**Review by Nikitzu — APPROVED** [review]

**Pull request merged**